lib/init.php

```diff
@@ -177,3 +177,4 @@ function register_site_icon_url( $response ) {
 add_filter( 'rest_index', 'register_site_icon_url' );
 
 add_theme_support( 'widgets-block-editor' );
+add_theme_support( 'block-templates' );
```

Comment:
@youknowriad I find this part a bit unexpected. Why are we unconditionally declaring theme support in the plugin/core? 🤔 

If we want to have this active by default, we could remove the theme support part altogether, and modify the activation checks to use the filter for opt-out, instead of:

> Theme authors can opt-out of this feature by removing the `block-templates` theme support in their `functions.php` file.

Reply:
This is something I explain here https://github.com/WordPress/gutenberg/pull/30465#discussion_r606174487

Instead of having a "disable-something" theme support, Core prefer to have theme supports enabled by default and plugins/themes can do `remove_theme_support`.

Reply:
Thanks for clarifying! While reading the activation conditions I assumed that this will be only enabled for themes that explicitly declare support, so I was a bit surprised afterward when I found out in testing that it actually works for all themes.

Reply:
yes, I the moment, this is opt-out because we want people to experience this, depending how the betas of WP go though, we might reconsider.

Reply:
To close the loop, #32858 makes it opt-in on classic themes and opt-out for themes using theme.json.